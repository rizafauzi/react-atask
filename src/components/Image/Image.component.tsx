import { useState } from 'react';

import { ReactComponent as ImageIcon } from '@assets/icons/image.icon.svg';
import { color } from '@styles/color';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import 'react-lazy-load-image-component/src/effects/blur.css';
import classnames from 'classnames';

interface ImageProps {
  alt: string;
  src: string;
  href?: string;
  width?: number;
  height?: number;
  quality?: number;
  priority?: boolean;
  className?: string;
}

const Image = ({
  alt,
  src,
  className,
  width = 50,
  height = 50,
}: ImageProps) => {
  const [isError, setIsError] = useState(false);

  if (isError) {
    return (
      <ImageIcon
        width={width}
        height={height}
        className="error-image"
        color={color.primary100}
      />
    );
  }

  return (
    <LazyLoadImage
      alt={alt}
      src={src}
      width={width}
      effect="blur"
      height={height}
      data-testid="image"
      onError={() => setIsError(true)}
      className={classnames(className, 'overflow-hidden')}
    />
  );
};

export default Image;
