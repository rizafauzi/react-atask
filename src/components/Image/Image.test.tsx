import { render, screen } from '@testing-library/react';

import Image from './Image.component';

test('should render Image component', () => {
  render(
    <Image
      width={30}
      height={30}
      quality={0.5}
      alt="rizafauzi"
      className="rounded-full"
      src="https://avatars.githubusercontent.com/u/46235432?v=4"
    />
  );
  const image = screen.getByTestId('image');
  expect(image).toBeInTheDocument();
});

test('matches snapshot', () => {
  const image = render(
    <Image
      width={30}
      height={30}
      quality={0.5}
      alt="rizafauzi"
      className="rounded-full"
      src="https://avatars.githubusercontent.com/u/46235432?v=4"
    />
  );
  expect(image).toMatchSnapshot();
});
