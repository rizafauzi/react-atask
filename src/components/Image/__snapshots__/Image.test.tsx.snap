// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class=" lazy-load-image-background blur"
        style="color: transparent; display: inline-block; height: 30px; width: 30px;"
      >
        <img
          alt="rizafauzi"
          class="rounded-full overflow-hidden"
          data-testid="image"
          height="30"
          src="https://avatars.githubusercontent.com/u/46235432?v=4"
          width="30"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class=" lazy-load-image-background blur"
      style="color: transparent; display: inline-block; height: 30px; width: 30px;"
    >
      <img
        alt="rizafauzi"
        class="rounded-full overflow-hidden"
        data-testid="image"
        height="30"
        src="https://avatars.githubusercontent.com/u/46235432?v=4"
        width="30"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
